src/Object/ObjectFactory.php

```diff
@@ -128,4 +120,24 @@ final protected function instantiate(array $parameters): array
 
         return [$object, $parameters];
     }
+
+    /**
+     * @param Attributes $attributes
+     *
+     * @internal
+     */
+    final protected function mergedAttributes(array|callable $attributes): array
+    {
+        $attributes = parent::mergedAttributes($attributes);
```

Comment:
I'm not found of this kind of thing...

I'm afraid we'd enter in a "inheritance" hell at some point (`UserFactory` extends `PersistenceObjectFactory` extends `ObjectFactory` extends `BaseFactory`)
I'm really wondering if we cannot find another design where we'd use composition over inheritance, but I cannot find a nice pattern :shrug: